src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2768,6 +2768,52 @@ public override BoundNode VisitNonConstructorMethodBody(BoundNonConstructorMetho
             return null;
         }
 
+        public override BoundNode VisitNullCoalesingAssignmentOperator(BoundNullCoalesingAssignmentOperator node)
+        {
+            LocalState savedState;
+            if (RegularPropertyAccess(node.LeftOperand))
+            {
+                var left = (BoundPropertyAccess)node.LeftOperand;
+                var property = left.PropertySymbol;
+                if (property.RefKind == RefKind.None)
+                {
+                    var readMethod = property.GetOwnOrInheritedGetMethod() ?? property.SetMethod;
+                    var writeMethod = property.GetOwnOrInheritedSetMethod() ?? property.GetMethod;
```

Comment:
>property.GetOwnOrInheritedSetMethod() ?? property.GetMethod [](start = 38, length = 59)

There is something strange here. Why are we putting `SetMethod` into `readMethod`, and `GetMethod` into `writeMethod`?

Reply:
Because I copied this from the compound assignment version, and that apparently has a bug :).

---
In reply to: [200785539](https://github.com/dotnet/roslyn/pull/28253#discussion_r200785539) [](ancestors = 200785539)

Reply:
Will follow up with @gafter on the version in VisitCompoundAssignmentOperator.

---
In reply to: [200788466](https://github.com/dotnet/roslyn/pull/28253#discussion_r200788466) [](ancestors = 200788466,200785539)